### PR TITLE
Add .git to excludes for ruff in pyproject.toml

### PR DESCRIPTION
**Pull request by schrockn**
### Summary & Motivation

Looks like ruff was parsing all the files in .git. Excluding it via pyproject.toml

### How I Tested These Changes

(dagster-x86-3.10.3) elementl-13in-m1-mbppro-2021:dagster schrockn$ make ruff

Before:

```
(dagster-x86-3.10.3) elementl-13in-m1-mbppro-2021:dagster schrockn$ make ruff
ruff --fix .
.git/logs/refs/remotes/origin/yuhan/12-12-convert-examples_4.2/_assets_pandas_type_metadata_repository.py_-___init__.py:1:43: E999 SyntaxError: invalid syntax. Got unexpected token 'a13b52de68dfa49916333202bf26d70ab50cb8db'
Found 1 error(s).
make: *** [ruff] Error 1
(dagster-x86-3.10.3) elementl-13in-m1-mbppro-2021:dagster schrockn$
```

After
```
(dagster-x86-3.10.3) elementl-13in-m1-mbppro-2021:dagster schrockn$ make ruff
ruff --fix .
(dagster-x86-3.10.3) elementl-13in-m1-mbppro-2021:dagster schrockn$
```

### CODE REVIEW

**Comment by schrockn** [comment]
Current dependencies on/for this PR:
* master
  * **PR #11534** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11534" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/11534?utm_source=stack-comment).

**Review by smackesey — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -86,13 +86,14 @@ exclude = [
   "*/__generated__/*",
```
Comment (charliermarsh) [inline pyproject.toml:86]: Sorry for the drive-by, but if you use `extend-exclude` instead of `exclude`, Ruff should preserve the "default" exclusions list by augmenting (rather than replacing) it:

```rust
[
    FilePattern::Builtin(".bzr"),
    FilePattern::Builtin(".direnv"),
    FilePattern::Builtin(".eggs"),
    FilePattern::Builtin(".git"),
    FilePattern::Builtin(".hg"),
    FilePattern::Builtin(".mypy_cache"),
    FilePattern::Builtin(".nox"),
    FilePattern::Builtin(".pants.d"),
    FilePattern::Builtin(".ruff_cache"),
    FilePattern::Builtin(".svn"),
    FilePattern::Builtin(".tox"),
    FilePattern::Builtin(".venv"),
    FilePattern::Builtin("__pypackages__"),
    FilePattern::Builtin("_build"),
    FilePattern::Builtin("buck-out"),
    FilePattern::Builtin("build"),
    FilePattern::Builtin("dist"),
    FilePattern::Builtin("node_modules"),
    FilePattern::Builtin("venv"),
]
```
